ngraph/core/src/op/random_uniform.cpp

```diff
@@ -0,0 +1,201 @@
+                          min_element_type == max_element_type,
+                          "'min_val' should have the same type as 'max_val'.");
+    NODE_VALIDATION_CHECK(
+        this,
+        min_element_type == get_out_type(),
+        "'min_val' and 'max_val' should have the same type as 'out_type' attribute.");
+
+    if (const auto& const_min = get_constant_from_source(input_value(1)))
+    {
+        if (const auto& const_max = get_constant_from_source(input_value(2)))
+        {
+            if (get_out_type() == ngraph::element::Type_t::i64 ||
+                get_out_type() == ngraph::element::Type_t::i32)
+            {
+                int64_t min_val = const_min->cast_vector<int64_t>()[0];
+                int64_t max_val = const_max->cast_vector<int64_t>()[0];
+
+                NODE_VALIDATION_CHECK(this,
+                                      min_val < max_val,
+                                      "Min value must be less than max value. Got "
+                                      "min value: ",
+                                      min_val,
+                                      ", max value: ",
+                                      max_val);
+            }
+            else
+            {
+                double min_val = const_min->cast_vector<double>()[0];
+                double max_val = const_max->cast_vector<double>()[0];
+
+                NODE_VALIDATION_CHECK(this,
+                                      min_val < max_val,
+                                      "Min value must be less than max value. Got "
+                                      "min value: ",
+                                      min_val,
+                                      ", max value: ",
+                                      max_val);
+            }
+        }
+    }
+
+    set_output_type(0, get_out_type(), output_shape);
+}
+
+bool op::v8::RandomUniform::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v8_RandomUniform_visit_attributes);
+    visitor.on_attribute("output_type", m_output_type);
+    visitor.on_attribute("seed", m_seed);
+    visitor.on_attribute("seed2", m_seed2);
+    return true;
+}
+
+shared_ptr<Node> op::v8::RandomUniform::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    NGRAPH_OP_SCOPE(v8_Roll_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return make_shared<v8::RandomUniform>(
+        new_args[0], new_args[1], new_args[2], m_output_type, m_seed, m_seed2);
+}
+
+bool op::v8::RandomUniform::evaluate(const HostTensorVector& outputs,
+                                     const HostTensorVector& inputs) const
+{
+    NGRAPH_OP_SCOPE(v8_Roll_evaluate);
+    const uint64_t* out_shape;
+    std::vector<uint64_t> out_shape_uint64;
+    out_shape_uint64.resize(shape_size(inputs[0]->get_shape()));
+
+    if (inputs[0]->get_element_type() == element::Type_t::u64)
+    {
+        out_shape = (uint64_t*)inputs[0]->get_data_ptr<const char>();
```

Comment:
Why not `inputs[0]->get_data_ptr<const uint64_t>()`

Reply:
You are right, it works with inputs[0]->get_data_ptr<const uint64_t>(). Corrected.